src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -4511,20 +4511,25 @@ and TcTupleType kindOpt (cenv: cenv) newOk checkConstraints occ env tpenv isStru
         else
             let argsR,tpenv = TcTypesAsTuple cenv newOk checkConstraints occ env tpenv args m
             TType_tuple(tupInfo, argsR), tpenv
+            
+and CheckAnonRecdTypeDuplicateFields (ident: _ -> Ident) elems = 
+    elems |> List.iteri (fun i uc1 -> 
+        elems |> List.iteri (fun j uc2 -> 
+            let id1 = (ident uc1)
+            let id2 = (ident uc2)
+            if j > i && id1.idText = id2.idText then 
+               errorR(Error(FSComp.SR.tcAnonRecdTypeDuplicateFieldId(id1.idText), id1.idRange))))
 
 and TcAnonRecdType (cenv: cenv) newOk checkConstraints occ env tpenv isStruct args m =
     let tupInfo = mkTupInfo isStruct
+    let idents = args |> List.map fst
```

Comment:
Is there a reason for moving this above the line declaring `unsortedFieldIds`? Because `idents` is basically the same thing as a list, no point in computing it twice I think (also, if you had an array instead of a list, you could just do a nested for loop instead of nested `iteri`, which IMO is slightly clearer).

Reply:
Yeah, make sense to use unsortedFieldIds. Updated now